### PR TITLE
[Server] Support for publishing to catalog

### PR DESCRIPTION
**Pull request by gr455**
Signed-off-by: Ruturaj Mohite <mohite.ruturaj15@gmail.com>

**Description**
Adds API on meshery server to publish content to catalog

**Notes for Reviewers**
Use the corresponding changes in meshery cloud for testing

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Review by leecalcote — CHANGES_REQUESTED** [review]
Missing Swagger / Meshery Docs update.

**Comment by gr455** [comment]
Did I do the swagger part right?

**Comment by gr455** [comment]
> Did I do the swagger part right?

The answer is 'no' :smiley:

**Comment by leecalcote** [comment]
> > Did I do the swagger part right?
> 
> The answer is 'no' 😃

😂

File: server/handlers/meshery_pattern_handler.go [hunk 1]
```diff
@@ -381,6 +381,43 @@ func (h *Handler) CloneMesheryPatternHandler(
 	fmt.Fprint(rw, string(resp))
 }
 
+// swagger:route POST /api/pattern/catalog/publish PatternsAPI idPublishCatalogPatternHandler
+//
```
Comment (theBeginner86) [inline server/handlers/meshery_pattern_handler.go:385]:
```suggestion
// Handle Publish for a Meshery Pattern
//
```

nitpick. 
Would fix this: 

Screenshot 2022-10-12 at 9 33 35 PM

File: server/handlers/meshery_filter_handler.go [hunk 2]
```diff
@@ -276,6 +276,43 @@ func (h *Handler) CloneMesheryFilterHandler(
 	fmt.Fprint(rw, string(resp))
 }
 
+// swagger:route POST /api/filter/catalog/publish FiltersAPI idPublishCatalogFilterHandler
+//
```
Comment (theBeginner86) [inline server/handlers/meshery_filter_handler.go:280]:
```suggestion
// Handle Publish for a Meshery Filter
//
```

File: server/handlers/meshery_pattern_handler.go (hunk 1 above)
Comment (gr455) [reply server/handlers/meshery_pattern_handler.go:385]: Whoops these comments slipped my mind. Thanks for reviewing @theBeginner86. Will fix these

**Review by theBeginner86 — APPROVED** [review]
LGTM!

**Pull request merged**